src/test/coins_tests.cpp

```diff
@@ -571,11 +571,19 @@ const static CAmount VALUE2 = 200;
 const static CAmount VALUE3 = 300;
 const static char DIRTY = CCoinsCacheEntry::DIRTY;
 const static char FRESH = CCoinsCacheEntry::FRESH;
-const static char NO_ENTRY = -1;
```

Comment:
Since we don't have pass a bitmap as flags anymore, does it make sense to get rid of the `char` type for flags in `coins_tests` and replace with a more descriptive enum?
```C++
enum Flags
{
   Clean,
   Dirty,
   Fresh,
   DirtyAndFresh
};
```

Reply:
what's the difference between clean and NO_ENTRY? Does that difference still make sense after this change?

> get rid of the char type for flags in coins_tests and replace with a more descriptive enum

Since we don't have flags in the production code anymore I would prefer getting rid of them in the tests as well. Reintroducing an enum would kinda' defeat that purpose in my opinion.

To make sure the tests are still checking the same combinations, I have added a few scripted diffs to transition the tests away from the flags - now the tabular data displays dirty and fresh booleans instead (I guess we can group some of the examples into `CoinMapEntry` as well, if needed).
This should help us selectively decide which combinations are still valid.
What do you think?

Reply:
`Clean` means not dirty and not fresh. `NO_ENTRY` is a special value that is not a possible flag but signals to the tests that that entry does not exist. With proper types, that would correspond to a `std::nullopt`.
We don't have to name the enum `Flags`, we can name it `EntryState`. That way we don't have to validate the `flags` variable to see if it's one of the values we will accept. That is better served by using an enum and std::optional.

Reply:
I don't think replacing with true and false is the right approach. Using enums, tests would be like
```
CheckAccessCoin(VALUE1, SPENT , SPENT , DirtyAndFresh, DirtyAndFresh);
CheckSpendCoins(ABSENT, SPENT , ABSENT, Fresh        , std::nullopt );
```

Reply:
> is a special value that is not a possible flag but signals to the tests that that entry does not exist

> That is better served by using an enum and std::optional.

Thanks, I'll investigate tomorrow.
Please leave some comments about the overall direction as well, do you think this new test layout is more readable or less?

Reply:
Did a significant restructure, the tabular tests should be a lot easier to comprehend and modify now.
I still need to get rid of remaining flags in the tests, will do that tomorrow - any early feedback until then is appreciated.

Reply:
Done